lib/src/revset.rs

```diff
@@ -2211,20 +2211,35 @@ impl PartialSymbolResolver for ChangePrefixResolver<'_> {
     }
 }
 
+/// An extension of the DefaultSymbolResolver.
+///
+/// Each PartialSymbolResolver will be invoked in order, its result used if one
+/// is provided. Native resolvers are always invoked first. In the future, we
+/// may provide a way for extensions to override native resolvers like tags and
+/// branches.
+pub trait SymbolResolverExtension {
+    fn new_resolvers<'a>(&self, repo: &'a dyn Repo) -> Vec<Box<dyn PartialSymbolResolver + 'a>>;
```

Comment:
Do we still let `PartialSymbolResolver` capture `repo: &'a`?

If I understand it, the new API allows some `repo`-dependent initialization in `new_resolvers()`, but the constructed resolver should use the `repo` passed later as an argument.